### ISSUE TITLE
New code doesn't work.

### ISSUE THREAD

**Issue by HelloSmithers** [imchlorine/MykiBalance#2]
Your new version doesn't work with concession Myki's. I have tested and get the attached errors. 
![asset](https://user-images.githubusercontent.com/74478544/110232931-05caa580-7f75-11eb-81e1-cd722148e78d.jpg)
![error](https://user-images.githubusercontent.com/74478544/110232934-06fbd280-7f75-11eb-9f0a-096d75c74b64.jpg)

It's a bit weird as the old version worked fine. Do you have the old code by any chance? I stupidly over-wrote the old code with the new code and now cannot downgrade.

**Comment by imchlorine** [comment]
Hi, did u follow the new instructions?  If it's still not working, would do mind send me ur Card number to my email address to help me address the issue?  imchlorine@gmail.com 
Or If u insist on using the old version, please go to the commit history here https://github.com/imchlorine/MykiBalance/blob/a1cb8188c6916ec531167912cf90a57d44160850/MykiBalance.js

**Comment by HelloSmithers** [comment]
I might just stick to the old version. Thanks for the help!

**Comment by bjeanes** [comment]
I'm hitting this too.

If I hit play inside Scriptable I get a different error:

```
0.mykiCardNumber: Expected a non-empty value. Got: null
```

I'll email you my Myki number for you to see if you can repro

**Comment by bjeanes** [comment]
I tried 3 other card numbers.

When I port your code to a simple `curl` request, I can see data.

**Edit: below is a red herring, because running script in app is not in context of widget so doesn't have the number via `args.widgetParameter`.**

----

I can reproduce the exact error I see in the logs by passing `null` for the card:

```
❯ curl -H 'x-ptvwebauth: 1624591892-wVOuseeq/JlGNabp9yv7f7Thu24NazTn3iWwL7v+neY=' -H 'Content-Type: application/json' -H 'Accept: application/json' -d '{"0": {"mykiCardNumber": null}}' "https://mykiapi.ptv.vic.gov.au/v2/myki/card"
{"code":409,"message":"0.mykiCardNumber: Expected a non-empty value. Got: null"}
```
<small>(Obviously that auth header will likely expire before anyone can simply copy it. I followed your code to acquire one and manually ran the request)</small>

It seems like perhaps the params are not actually being passed correctly from the widget to the script. I don't yet know enough about how Scriptable works to understand why that might be, though.
